lib/ansible/modules/windows/win_package.ps1

```diff
@@ -1253,77 +1258,62 @@ namespace Source
+$name = Get-AnsibleParam -obj $params -name "name" -type "str" -default $path
+$productid = Get-AnsibleParam -obj $params -name "productid" -type "str" -aliases "product_id"
+$arguments = Get-AnsibleParam -obj $params -name "arguments" -type "str"
+$state = Get-AnsibleParam -obj $params -name "state" -type 'str' -default "present" -validateset "absent","present" -aliases "ensure"
+$username = Get-AnsibleParam -obj $params -name "username" -type "str" -aliases "user_name"
+$password = Get-AnsibleParam -obj $params -name "password" -type "str" -aliases "user_password"
+$return_code = Get-AnsibleParam -obj $params -name "expected_return_code" -type "list" -default @(0, 3010)
 
 $result = @{
+    arguments = $arguments
     changed = $false
+    failed = $false
+    name = $name
+    path = $path
+    productid = $productid
+    rc = $null
+    restart_required = $false
 }
 
-$path = Get-Attr -obj $params -name path -failifempty $true -resultobj $result
-$name = Get-Attr -obj $params -name name -default $path
-$productid = Get-Attr -obj $params -name productid
-if ($productid -eq $null)
-{
-    #Alias added for backwards compat.
-    $productid = Get-Attr -obj $params -name product_id -failifempty $true -resultobj $result
-}
-$arguments = Get-Attr -obj $params -name arguments
-$ensure = Get-Attr -obj $params -name state -default "present"
-if ($ensure -eq $null)
-{
-    $ensure = Get-Attr -obj $params -name ensure -default "present"
-}
-$username = Get-Attr -obj $params -name user_name
-$password = Get-Attr -obj $params -name user_password
-$return_code = Get-Attr -obj $params -name expected_return_code -default 0
-
-#Construct the DSC param hashtable
+# Construct the DSC param hashtable
 $dscparams = @{
-    name=$name
-    path=$path
-    productid = $productid
-    arguments = $arguments
-    ensure = $ensure
-    returncode = $return_code
+    Arguments = $arguments
+    Ensure = $state
+    Name = $name
+    Path = $path
+    Productid = $productid
+    Returncode = $return_code
+    WhatIf = $check_mode
 }
 
-if (($username -ne $null) -and ($password -ne $null))
-{
-    #Add network credential to the list
+if ($username -and $password) {
+    # Add network credential to the list
     $secpassword = $password | ConvertTo-SecureString -AsPlainText -Force
     $credential = New-Object pscredential -ArgumentList $username, $secpassword
-    $dscparams.add("Credential",$credential)
+    $dscparams.add("Credential", $credential)
 }
 
-#Always return the name
-$result.name = $name
-
-$testdscresult = Test-TargetResource @dscparams
-if ($testdscresult -eq $true)
-{
-    Exit-Json -obj $result
-}
-Else
-{
-    try
-    {
-        set-TargetResource @dscparams
-    }
-    catch
-    {
+if (-not (Test-TargetResource @dscparams)) {
+    try {
+        Set-TargetResource @dscparams
+    } catch {
         $errormsg = $_
-        Fail-Json -obj $result -message $errormsg.ToString()
+        Fail-Json -obj $result -message "This error $errormsg"
```

Comment:
Yes, need to figure out where in the code it's doing -WhatIf twice, because I do not understand it.